uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeavePeriodEntitlementTest.php

```diff
@@ -430,8 +430,8 @@ public function testCanSaveALeavePeriodEntitlementFromAnEntitlementCalculation()
     $this->assertEquals($type->id, $periodEntitlement->type_id);
     $this->assertEquals($this->contract['contact_id'], $periodEntitlement->contact_id);
 
-    // 10 + 1 + 3 (Pro Rata + Brought Forward + No. Public Holidays)
-    $this->assertEquals(14, $periodEntitlement->getEntitlement());
+    // 10 + 1 (Pro Rata (Including the Public Holidays) + Brought Forward)
+    $this->assertEquals(11, $periodEntitlement->getEntitlement());
```

Comment:
Off the top of my head, I don't know why this value should change from 14. The public holiday array contained 3 public holidays and now it was replaced with the integer 3 since we now need the number of public holidays. except something changed in the logic which I can't seem to see.
Shouldn't it still be 14?

Reply:
It changed because the number of public holidays is already included in the Pro Rata (as the comment says).

The previous version of the test was wrong. It uses an entitlement calculation mock that didn't have all the methods used by the `saveFromCalculation()` method. For saving the Leave amount, the `saveFromCalculation()` calls the `saveLeaveBalanceChange()` method, which calculates the amount with this:

```php
'amount' => $calculation->getProRata() - $calculation->getNumberOfPublicHolidaysInEntitlement()
```

The `getNumberOfPublicHolidaysInEntitlement()` method was not mocked, so it was actually running a DB query to get the number. Since this test doesn't save any Public Holidays in the database, it would return 0 rather than 3, and the store amount was wrong. 

After the changes, we now have a mock for the `getNumberOfPublicHolidaysInEntitlement()`, so the test started failing and the fix was to change the value from 14 to 11.

Reply:
Thanks for the explanation.
The number of public holidays in the test still seems confusing though because its difficult to imagine how the public holiday is part of the proRata, The way it is, if we change the `$numberOfPublicHolidays` to any value it will not affect the test outcome.
I would suggest that we change the `getEntitlementCalculationMock` function a little, Instead of the fifth parameter being the `proRata`, It will be `proRataWithoutPublicHoliday`, then in the function itself, we make the following changes

```php
    $calculation->expects($this->once())
                ->method('getProRata')
                ->will($this->returnValue($proRataWithoutPublicHoliday + $numberOfPublicHolidays));
```
This way it will be easy to see how the public holiday is reflected in the calculation of the overall proRata in the test.